rules/dynamic_archive.bzl

```diff
@@ -0,0 +1,215 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Dynamic archive macro loads repos from archives with dynamic tarballs but
+stable contents for OpenTitan.
+"""
+
+load(
+    "@bazel_tools//tools/build_defs/repo:utils.bzl",
+    "patch",
+    "read_netrc",
+    "read_user_netrc",
+    "update_attrs",
+    "use_netrc",
+    "workspace_and_buildfile",
+)
+
+def _get_auth(ctx, urls):
+    """Given the list of URLs obtain the correct auth dict."""
+    if ctx.attr.netrc:
+        netrc = read_netrc(ctx, ctx.attr.netrc)
+    else:
+        netrc = read_user_netrc(ctx)
+    return use_netrc(netrc, urls, ctx.attr.auth_patterns)
+
+def _check_dir_sha(ctx):
+    """Utility function checks the sha256 of a directory instead of an archive
+
+    It assumes the parameters `sha256` and `output` to be present in
+    `ctx.attr`; the latter possibly with value None.
+
+    Args:
+      ctx: The repository context of the repository rule calling this utility
+      function.
+
+    Returns the sha256 of a gzipped tarball of the directory contents.
+    """
+    command = "find . -type f -exec sha256sum {} \\; | sort -k 2 | sha256sum"
+    st = ctx.execute(["bash", "-c", command])
+    if st.return_code:
+        fail("Error listing and hashing with command %s:\n%s$s" %
+             (command, st.stdout, st.stderr))
+    else:
+        sha256 = st.stdout[:64]
+    if ctx.attr.sha256 and ctx.attr.sha256 != sha256:
+        tarv = ctx.execute(["bash", "-c", "find --version"])
```

Comment:
Did you mean `tar --version` here?

These all seem like debug prints for figuring out what is going wrong in CI.  I assume they'll be removed when you figure out what's going wrong.

Reply:
They were debug prints and I did leave them messy, I'll clean them up before committing.